app.js

```diff
@@ -21,4 +21,35 @@ export default async function app (fastify, opts) {
     dir: path.join(import.meta.dirname, 'routes'),
     options: { ...opts }
   })
+
+  fastify.setErrorHandler((err, request, reply) => {
+    request.log.error({
+      err,
+      request: {
+        method: request.method,
+        url: request.url,
+        query: request.query,
+        params: request.params
+      }
+    }, 'Unhandled error occurred')
+
+    reply.code(err.statusCode ?? 500)
```

Comment:
This is a nullish coalescing operator, it only returns 500 if `statusCode` is null. In the case it's just an empty string or a boolean, it will return that value. This is maybe want you want, or maybe not. Just saying.. I use `||` which is an or operator, that returns 500 if `statusCode` is "falsely".

Reply:
Okay, I tend to think that people always use a logicalOr expression check out of habit.
I was considering relying on the expected type here:
```ts
FastifyError.statusCode?: number | undefined
```